Add dictionary definition for "API"

Kushian01100111 opened the pull request:
## This PR fixes...

The lack of dictionary definition for Application Programming Interface(API)

## What I did...

- Add dictionary definition for API

## How to test...

_In a list format, tell us how to test this PR._
Ex:

1. Navigate to /dictionary
1. Search "Application Programming Interface(API)"
1. See if the definition appears under the term "Application Programming Interface(API)"

## I learned...
A deeper understanding of the APIs as a whole.

Kushian01100111 committed:
Add dictionary definition for "API"

novellac commented on src/assets/content/dictionary/application-programming-interface-api.md:
```diff
@@ -1,6 +1,11 @@
 ---
 title: Application Programming Interface (API)
-definition: ''
+definition: 'The word Application refers to any software with a distinct function, and interface can be thought as a contract of service between two applications.This contract defines how the two communicate with each other using requests and responses. 
+
+Then a API sets contracts with defined rules that explain how computers or applications communicate with one another.This allows services and products to communicate with each other and leverage each other's data and functionality through a documented interface.'.
```
```suggestion
definition: "The word Application refers to any software with a distinct function, and interface can be thought as a contract of service between two applications. This contract defines how the two communicate with each other using requests and responses. 

Then a API sets contracts with defined rules that explain how computers or applications communicate with one another. This allows services and products to communicate with each other and leverage each other's data and functionality through a documented interface."
```
bug: Since you have a single quote `'` inside the text, you'll need to use double quotes `"` to wrap the whole definition. I've also suggested to add spaces between the sentences for clarity. You can accept the suggestion via the Github UI, or you can make an extra commit to this PR.

Kushian01100111 replied:
Thank you for your suggestions, I commit them and added space for better readability.

novellac reviewed (changes requested):
Almost there! Unbreak the build with the suggested corrections, and I'll be happy to approve and merge.

Kushian01100111 committed:
Update src/assets/content/dictionary/application-programming-interface-api.md

Thank you for the suggestions, i am not totally sure how to add more changes to this correction.

Co-authored-by: novellac <38117965+novellac@users.noreply.github.com>

Kushian01100111 committed:
Add space between sentences

I added space between sentences for better readability

novellac reviewed (approved):
Thanks for the Hacktoberfest contribution! 🍁
